classes/security/Validation.php

```diff
@@ -474,6 +482,65 @@ public static function canAdminister($administeredUserId, $administratorUserId)
         // There were no conflicting roles. Permit administration.
         return true;
     }
+
+    /**
+     * Get the user's administration level
+     *
+     * @param int   $administeredUserId     User ID of user to potentially administer
+     * @param int   $administratorUserId    User ID of user who wants to do the administrating
+     * 
+     * @return int The authorized adminstration level 
+     */
+    public static function getAdministrationLevel($administeredUserId, $administratorUserId)
+    {
+        $roleDao = DAORegistry::getDAO('RoleDAO'); /** @var RoleDAO $roleDao */
+        $context = Application::get()?->getRequest()?->getContext();
```

Comment:
This binds the `Validation` class to the `Request`, which presumes web-based invocation. I'd rather not bind these together if we can avoid it, in case e.g. we need to call this from a command line tool or a queued job. I'd suggest passing the `$contextId` in as an optional 3rd variable.

Reply:
done .